frontend/public/components/environment.jsx

```diff
@@ -304,7 +308,17 @@ export const EnvironmentPage = connect(stateToProps)(
     }
 
     componentDidUpdate(prevProps) {
-      const { obj, model, impersonate, readOnly } = this.props;
+      const { obj, model, impersonate, readOnly, rawEnvData } = this.props;
+      const { dirty, currentEnvVars } = this.state;
+
+      // Underlying resource env vars have changed and the changes did not come from the last save action,
+      if (!_.isEqual(prevProps.rawEnvData, rawEnvData) && !_.isEqual(currentEnvVars.rawEnvData, rawEnvData)) {
```

Comment:
I'd think we could skip this check if the form is not dirty.

I haven't looked at this code for a while. Why do we need to check `rawEnvData` against two different objects?

Reply:
I don't think we can skip this check. We need to know if the env vars changed in the background no matter what state the form is in (dirty or not). If it's dirty, we just show a stale notification so the user can reload. If not, we just update the form to show the latest.

I think you're right about the two equality checks though, I can probably just check against the current state.

Reply:
If it's not dirty, I assumed we'd always update, although I guess that adds unnecessary re-renders.

Reply:
Yeah, I see what you mean now. The problem is that the state update has to have a condition around it, or else it will cause an infinite loop.